### ISSUE TITLE
twitch https update = html5 player (http) is unable to load via the iframe

### ISSUE THREAD

**Issue by onetimesignup** [EchoDev/TwitchHTML5#19]
just wanted to report it, you can reach the html5 player page if you update it to use https but videos refuse to start..

maybe it's the waiting game now, that twitch will update the html5 player page next

**Comment by EchoDev** [comment]
I just updated the script a few minutes ago. Should be working fine again.

**Comment by onetimesignup** [comment]
still not loading for me

**Comment by vladkosi** [comment]
Doesn't work in firefox.

Try https://player.twitch.tv/?html5&branding=false&showInfo=false&channel=esl_csgo

Blocked loading mixed active content "http://video50.fra01.hls.ttvnw.net/hls50/esl_csgo_20072465424_414333848/chunked/py-index-live.m3u8"

**Comment by EchoDev** [comment]
Hmm..That might be because Firefox (and Chrome probably too) doesn't allow http when a site is https.

In Firefox you can fix this by changing the following in about:config

security.mixed_content.block_active_content set to false

The videostream domain do not have HTTPS and Twitch seems to be redirecting the site itself automaticly to HTTPS.

**Comment by onetimesignup** [comment]
yea im pretty sure its just a matter of time before they push the https update to the html5 player page as well, after all the https just dropped for the regular page

**Comment by EchoDev** [comment]
It's not per-se a problem with the HTTPS player itself but the domain where the audio and video files come from. In console you can see the links that end with .m3u8 and .ts

Those are all HTTP. If you try to change them to HTTPS they just give you an error because they have no HTTPS set up for them. And Firefox and Chrome do not allow HTTP when HTTPs is active.

**Comment by onetimesignup** [comment]
is it possible for the script to look for which browser it's installed on then go inside about:config check bool value for security.mixed_content.block_active_content then change it to false if it's set to true ?
ugg its not a pretty solution, let me try again; is it possible to whitelist/add twitch.tv or something using the script ?

**Comment by EchoDev** [comment]
It might be possible to change settings but that is pretty scummy because it will weaken browser security. This might be a temporary solution, I left Twitch a comment @ reddit, hopefully they will read it and respond it. If they are still working on the HTML5 player they will also run into the same problem.

**Comment by onetimesignup** [comment]
>That's somewhat addressed in the FAQ. HLS content isn't over HTTPS yet but will have to be before the HTML5 player is released.

welp

**Comment by EchoDev** [comment]
I wonder if they will give an ETA for the HTML5 player and I hope they didn't forget HTTPs will add a fuckton of load on the servers lol...

**Comment by onetimesignup** [comment]
what if we open it as a popup but attatch it to the player div ? possible ?

**Comment by EchoDev** [comment]
Nope, pop up would still be https. The problem is the stream being http. Since Twitch is forcing the site to go to https it is also not possible use http for the player. Users NEED to have mixed content blocking disabled untill Twitch enables https on those domains.

**Comment by EchoDev** [comment]
Untill Twitch starts using HTTPS on their streaming domains there is no easy way around this except for allowing mixed content.
